File: app/frontend/src/types/FuelConditions.ts
export interface ForecastHour {
    time: string;
    temperature: number;
    humidity: number;
    precipitation: number;
    wind_speed: number;
    wind_direction: number;
    wind_gusts: number;
}

export interface WindShift {
    in_hours: number;
    from_deg: number;
    to_deg: number;
    delta_deg: number;
    message: string;
}

export interface FuelConditions {
    // current conditions
    temperature: number;
    humidity: number;
    wind_speed: number;
    wind_direction: number;
    wind_gusts: number;

    // Lowveld Fire Danger Index
    fdi: number;
    fdi_band: string;
    fdi_color: string;

    // 30-day dryness 
    kbdi: number;
    kbdi_series: number[];
    days_since_rain: number;
    rain_30d_mm: number;
    last_rain_mm: number;

    dryness_grass: number;
    dryness_shrub: number;
    dryness_plantation: number;

    forecast: ForecastHour[];
    wind_shift: WindShift | null;

    observed_at: string;
    source: string;
}

File: app/backend/src/schemas/fuel_conditions.py
from pydantic import BaseModel

class ForecastHour(BaseModel):
    time: str
    temperature: float
    humidity: float
    precipitation: float
    wind_speed: float
    wind_direction: float
    wind_gusts: float

class WindShift(BaseModel):
    in_hours: int
    from_deg: float
    to_deg: float
    delta_deg: float
    message: str

class FuelConditions(BaseModel):
    temperature: float
    humidity: float
    wind_speed: float
    wind_direction: float
    wind_gusts: float

    fdi: float
    fdi_band: str
    fdi_color: str

    kbdi: float
    kbdi_series: list[float]
    days_since_rain: int
    rain_30d_mm: float
    last_rain_mm: float

    dryness_grass: float
    dryness_shrub: float
    dryness_plantation: float

    forecast: list[ForecastHour]
    wind_shift: WindShift | None = None

    observed_at: str
    source: str = "open-meteo"
